pylint/extensions/_check_docs_utils.py

```diff
@@ -58,10 +58,19 @@ def possible_exc_types(node):
     """
     excs = []
     if isinstance(node.exc, astroid.Name):
-        excs = [node.exc.name]
+        inferred = safe_infer(node.exc)
+        if inferred:
+            excs = [inferred.name]
     elif (isinstance(node.exc, astroid.Call) and
           isinstance(node.exc.func, astroid.Name)):
-        excs = [node.exc.func.name]
+        target = safe_infer(node.exc.func)
+        if isinstance(target, astroid.ClassDef):
+            excs = [target.name]
+        elif isinstance(target, astroid.FunctionDef):
+            for ret in target.nodes_of_class(astroid.Return):
```

Comment:
Make sure that ret belongs to the same function, nodes_of_class might return from inner functions as well.